src/HatsSignerGateBase.sol

```diff
@@ -181,30 +181,26 @@ abstract contract HatsSignerGateBase is BaseGuard, SignatureDecoder, HatsOwnedIn
         }
     }
 
-    /// @notice Tallies the number of existing `safe` owners that wear a signer hat, sets `signerCount` to that value, and updates the `safe` threshold if necessary
+    /// @notice Tallies the number of existing `safe` owners that wear a signer hat and updates the `safe` threshold if necessary
     /// @dev Does NOT remove invalid `safe` owners
     function reconcileSignerCount() public {
-        address[] memory owners = safe.getOwners();
-        uint256 validSignerCount = _countValidSigners(owners);
+        uint256 signerCount = validSignerCount();
 
-        if (validSignerCount > maxSigners) {
+        if (signerCount > maxSigners) {
             revert MaxSignersReached();
         }
 
-        // update the signer count accordingly
-        signerCount = validSignerCount;
-
         uint256 currentThreshold = safe.getThreshold();
         uint256 newThreshold;
         uint256 target = targetThreshold; // save SLOADs
 
-        if (validSignerCount <= target && validSignerCount != currentThreshold) {
-            newThreshold = validSignerCount;
-        } else if (validSignerCount > target && currentThreshold < target) {
+        if (signerCount <= target && signerCount != currentThreshold) {
+            newThreshold = signerCount;
+        } else if (signerCount > target && currentThreshold < target) {
```

Comment:
This would mean that if you're ever able to get `target >= currentThreshold`, then if `signerCount > target`, `newThreshold` will never be set, so it'll end up setting it to zero.